presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowBlockBuilder.java

```diff
@@ -0,0 +1,945 @@
+            }
+        }
+        return builder.build();
+    }
+
+    public Block buildBlockFromTimeStampMilliVector(TimeStampMilliVector vector, Type type)
+    {
+        if (!(type instanceof TimestampType)) {
+            throw new IllegalArgumentException("Expected TimestampType but got " + type.getClass().getName());
+        }
+
+        BlockBuilder builder = type.createBlockBuilder(null, vector.getValueCount());
+        for (int i = 0; i < vector.getValueCount(); i++) {
+            if (vector.isNull(i)) {
+                builder.appendNull();
+            }
+            else {
+                long millis = vector.get(i);
+                type.writeLong(builder, millis);
+            }
+        }
+        return builder.build();
+    }
+
+    public Block buildBlockFromFloat8Vector(Float8Vector vector, Type type)
+    {
+        BlockBuilder builder = type.createBlockBuilder(null, vector.getValueCount());
+        for (int i = 0; i < vector.getValueCount(); i++) {
+            if (vector.isNull(i)) {
+                builder.appendNull();
+            }
+            else {
+                type.writeDouble(builder, vector.get(i));
+            }
+        }
+        return builder.build();
+    }
+
+    public Block buildBlockFromFloat4Vector(Float4Vector vector, Type type)
+    {
+        BlockBuilder builder = type.createBlockBuilder(null, vector.getValueCount());
+        for (int i = 0; i < vector.getValueCount(); i++) {
+            if (vector.isNull(i)) {
+                builder.appendNull();
+            }
+            else {
+                int intBits = Float.floatToIntBits(vector.get(i));
+                type.writeLong(builder, intBits);
+            }
+        }
+        return builder.build();
+    }
+
+    public Block buildBlockFromVarBinaryVector(VarBinaryVector vector, Type type)
+    {
+        BlockBuilder builder = type.createBlockBuilder(null, vector.getValueCount());
+        for (int i = 0; i < vector.getValueCount(); i++) {
+            if (vector.isNull(i)) {
+                builder.appendNull();
+            }
+            else {
+                byte[] value = vector.get(i);
+                type.writeSlice(builder, Slices.wrappedBuffer(value));
+            }
+        }
+        return builder.build();
+    }
+
+    public Block buildBlockFromVarCharVector(VarCharVector vector, Type type)
+    {
+        if (!(type instanceof VarcharType)) {
+            throw new IllegalArgumentException("Expected VarcharType but got " + type.getClass().getName());
+        }
+
+        BlockBuilder builder = type.createBlockBuilder(null, vector.getValueCount());
+        for (int i = 0; i < vector.getValueCount(); i++) {
+            if (vector.isNull(i)) {
+                builder.appendNull();
+            }
+            else {
+                String value = new String(vector.get(i), StandardCharsets.UTF_8);
+                type.writeSlice(builder, Slices.utf8Slice(value));
```

Comment:
Can you create a `Slice` directly from the raw UTF-8 bytes from `vector.get(i)` using `Slice wrappedBuffer(byte... array)`? That would be more efficient that copying to a String first.